packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```diff
@@ -380,6 +393,37 @@ export const useCreditCard = props => {
      * Effects
      */
 
+    const hasNonce = !!(paymentNonceData && paymentNonceData.cart.paymentNonce);
+    const selectedMethodIsBraintree =
+        paymentMethodData &&
+        paymentMethodData.cart.selected_payment_method.code === 'braintree';
+
+    // On first render we should wait to know if we have previously saved data.
+    // If so, we can just tell the parent component to display the summary or
+    // go ahead and render the CC form.
+    useEffect(() => {
+        if (paymentMethodData && paymentNonceData) {
+            if (!shouldRender) {
+                // It's possible to get here with saved data but not be the
+                // selected method if a user enters CC and then changes payment
+                // method. In that case we should just re-render the CC form
+                // to have the user "re-select" the braintree payment method.
+                if (selectedMethodIsBraintree && hasNonce) {
+                    setDoneEditing(true);
```

Comment:
I just realized that this will get called by the credit card mounted within the edit modal. It's not really that important because by this time we're already displaying the summary view but in the future it could represent an issue.

I discussed this in a response to a thread here but this is one of two ways I can see that we can handle refresh/revisit logic.

I guess another alternative would be to never store nonce in cache and to always force users to re-enter CC data on refresh/revisit of checkout page.

Reply:
Oops this is actually an issue - now it never renders in modal if you've got a saved CC :(